src/ODE4.jl

```diff
@@ -0,0 +1,83 @@
+"""
+        In this file we want to estimate parameters of an ODE model describing the number of cells in G1 or G2 phase of the cell cycle 
+"""
+
+
+""" Time- and state-invariant model Jacobian. """
+function update_coef4(A, u, p, t)
+    A[:, :] .= [-p[1] .- p[5] 0.0 0.0 2.0*p[4]; p[1] -p[2] .- p[5] 0.0 0.0; 0.0 p[2] -p[3] .- p[6] 0.0; 0.0 0.0 p[3] -p[4] .- p[6]]
+end
+
+
+""" Predicts the model given a set of parametrs. """
+function predict(p, g1_0, g2_0, i, t)
+    u0 = 0.5*[g1_0[i], g1_0[i], g2_0[i], g2_0[i]]
+    A = zeros(eltype(p), 4, 4)
+    update_coef4(A, nothing, p, nothing)
+    Op = DiffEqArrayOperator(A, update_func=update_coef4)
+    prob = ODEProblem(Op, u0, extrema(t), p)
+    solution = solve(prob, AutoTsit5(Rosenbrock23()))
+    return prob, solution
+end
+
+""" Calculates the cost function for a given set of parameters. """
+function cost(p, g1_0, g2_0, g1, g2, i)
+    t = range(0.0; stop = 95.5, length = 192)
+    _, solution = predict(p, g1_0, g2_0, i, t)
+    res = zeros(2, 192)
+    G1 = solution(t, idxs=1).u + solution(t, idxs=2).u
+    G2 = solution(t, idxs=3).u + solution(t, idxs=4).u
+    res[1, :] = (G1 - g1[:, i]).^2
+    res[2, :] = (G2 - g2[:, i]).^2
+    summ = sum(res[1,:]) + sum(res[2,:])
+    return summ
+end
+
+""" Fit the ODE model to data. """
+function ODEoptimizer4(lower_bound::Array, upper_bound::Array, p::Array, i::Int, g1::Matrix, g2::Matrix, g1_0::Array, g2_0::Array)
```

Comment:
You shouldn't need to change the lower and upper bounds for the problem, so I'd set them in the function here, rather than pass them as arguments.

Reply:
done.